### PR TITLE
Create BndStrMPoleSymplectic4RadPass

### PR DESCRIPTION
**Pull request by catohxb**
Hung-Chun Chao created BndStrMPoleSymplectic4RadPass to include synchrotron radiation damping for the straight-geometry bending magnet.

### CODE REVIEW

**Review by lfarv — APPROVED** [review]
Nice addition

**Comment by lfarv** [comment]
Any other comment? For me it's ok for merging.

**Comment by swhite2401** [comment]
Ok for me!

File: atintegrators/BndStrMPoleSymplectic4RadPass [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+        /* ALLOCATE memory for the output array of the same size as the input */
+        plhs[0] = mxDuplicateArray(prhs[1]);
+        r_in = mxGetDoubles(plhs[0]);
+        BndStrMPoleSymplectic4RadPass(r_in, flen, irho, PolynomA, PolynomB,
+            MaxOrder, NumIntSteps, EntranceAngle, ExitAngle, X0ref, ByError,
+            RefDZ, Fint1, Fint2, Gap, T1, T2, R1, R2, Energy, num_particles);
+    }
+    else if (nrhs == 0) {
+        /* list of required fields */
+	    plhs[0] = mxCreateCellMatrix(8,1);
```
Comment (carmignani) [inline atintegrators/BndStrMPoleSymplectic4RadPass:401]: Hello,
it's a small detail, but this should be  plhs[0] = mxCreateCellMatrix(9,1);

File: atintegrators/BndStrMPoleSymplectic4RadPass [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+	    mxSetCell(plhs[0],1,mxCreateString("BendingAngle"));
+	    mxSetCell(plhs[0],2,mxCreateString("EntranceAngle"));
+	    mxSetCell(plhs[0],3,mxCreateString("ExitAngle"));
+        mxSetCell(plhs[0],4,mxCreateString("PolynomA"));
+	    mxSetCell(plhs[0],5,mxCreateString("PolynomB"));
+	    mxSetCell(plhs[0],6,mxCreateString("MaxOrder"));
+	    mxSetCell(plhs[0],7,mxCreateString("NumIntSteps"));
+        mxSetCell(plhs[0],8,mxCreateString("Energy"));
+	    if(nlhs>1) {    /* list of optional fields */
+	        plhs[1] = mxCreateCellMatrix(7,1);
```
Comment (carmignani) [inline atintegrators/BndStrMPoleSymplectic4RadPass:412]: And this should be plhs[1] = mxCreateCellMatrix(10,1);

**Review by carmignani — CHANGES_REQUESTED** [review]
There are two small modifications to do in the mexfunction.

File: atintegrators/BndStrMPoleSymplectic4RadPass (hunk 2 above)
Comment (chaohc) [reply atintegrators/BndStrMPoleSymplectic4RadPass:412]: Thank you for the catch!

**Comment by swhite2401** [comment]
Anyone implementing these changes?

**Comment by carmignani** [comment]
I could do the change, but I'm not sure to be able to do a commit on the forked version of AT from @catohxb. I can do the change by creating a branch in the atcollab official AT and then a new pull request. I will do that tomorrow, unless @catohxb will change his version before.

**Comment by carmignani** [comment]
Ok, actually I could do the commit directly online from gitHub on the forked version.

**Review by carmignani — APPROVED** [review]

**Comment by swhite2401** [comment]
@carmignani  If you add me as reviewer I can re-approve (needed after your changes and cannot be you)

**Review by swhite2401 — APPROVED** [review]

**Pull request merged**